test/GLMTest.m

```diff
@@ -22,6 +22,178 @@ function tear_down(testCase)
     
     methods(Test)
         
+        function test_ols_svd(testCase)
+            dt = 1; %sec
+            nb_samples = 700;
+            nb_channels=10;
+            nb_active_channels=5;
+            
+            [X, names,tmp] = design_matrix(0:dt:(nb_samples-1));
+
+            
+            [B, active_channels] = create_B_map(nb_channels,nb_active_channels);
+            y=X*B + 1e-3*randn(nb_samples,nb_channels);
+            
+            [B_hat,proj_X] = process_nst_glm_fit('compute_B_svd',y,X);
+            testCase.assertTrue( all(all(abs(B - B_hat) < 1e-3)));
+        end
+        
+        function test_AR_ols_fit(testCase)
+            dt = 1; %sec
+            nb_samples = 700;
+            nb_channels=10;
+            nb_active_channels=5;
+            
+            [X, names,tmp] = design_matrix(0:dt:(nb_samples-1));
+
+            
+            [B, active_channels] = create_B_map(nb_channels,nb_active_channels);
+            y=X*B + 1e-3*randn(nb_samples,nb_channels);
+            
+            [B_hat, covB, dfe, residuals, mse_residuals] = process_nst_glm_fit('AR1_ols_fit',y, dt, X, 0.1);
+            testCase.assertTrue( all(all(abs(B - B_hat) < 1e-3)));  
+         end
+
+        function test_channel_simulation_SNR_5(testCase)
+            
+            method_test=2; % 1 is to test pre-coloring, 2 is to test pre-whitening 
+            
+            repo_url = nst_get_repository_url();
+            data_fns = nst_request_files({{'unittest','nirscomp_data','GraspingRight_Hb_baseline_#1_.nirs'}}, ...
```

Comment:
Could you send me this file by email so that I put it only on the nirstorm data repo ?